app/assets/javascripts/admin/views/broken-links-report.js

```diff
@@ -0,0 +1,60 @@
+window.GOVUK = window.GOVUK || {}
+window.GOVUK.Modules = window.GOVUK.Modules || {};
+
+(function (Modules) {
+  function BrokenLinksReport (module) {
+    this.module = module
+  }
+
+  BrokenLinksReport.prototype.init = function () {
+    if (!fetch) return
+    var button = this.module.querySelector('button')
+    var authenticityTokenField = this.module.querySelector('[name="authenticity_token"]')
+    if (button && authenticityTokenField) this.setupEventListener(button, authenticityTokenField.value)
+    var refreshLink = this.module.querySelector('.js-broken-links-refresh')
+    if (refreshLink) {
+      this.setupPolling(refreshLink.href)
+      refreshLink.remove()
+    }
+  }
+
+  BrokenLinksReport.prototype.setupEventListener = function (button, authenticityToken) {
+    var action = this.module.querySelector('form').action
+    button.addEventListener('click', function (event) {
+      fetch(action + '.json', {
+        method: 'POST',
+        headers: { 'Content-Type': 'application/json' },
+        body: JSON.stringify({ authenticity_token: authenticityToken })
+      }).then(function (response) { return response.json() })
+        .catch(function () { location.reload() })
+        .then(function (json) {
+          this.replaceContentsAndCopyClasses(json.html)
+          this.init()
+        }.bind(this))
+      event.preventDefault()
+    }.bind(this))
+  }
+
+  BrokenLinksReport.prototype.setupPolling = function (href) {
+    GOVUK.smartPoller(2000, function (retry) {
+      fetch(href + '.json').then(function (response) { return response.json() })
+        .then(function (json) {
+          if (json.inProgress) {
+            retry()
+          } else {
+            this.replaceContentsAndCopyClasses(json.html)
+            this.init()
+          }
+        }.bind(this))
+        .catch(function () { location.reload() })
```

Comment:
I don't think a reload is a good thing to do here because the user may be in the middle of editing something and would suddenly find the page abruptly refreshing.

Reply:
The logic behind refreshing is that that's the default behaviour without javascript.
The button should only but present on the summary page so no changes should be lost.
What do you think?
Happy to keep or remove it, it feel quite unlikely that the promise will fail.

Reply:
I think we notify the user that an error occurred rather than refreshing. It could be that we display a message saying an error occurred and then give them a button to refresh the page, but it should be a user decision.

I hope that makes sense.

Reply:
As discussed I've updated the code to re-show the refresh button in the case that we fail 10 times and updated this to simply call `retry()`